pkg/apiserver/profiling/service.go

```diff
@@ -145,6 +149,43 @@ func (s *Service) serviceLoop(ctx context.Context) {
 	}
 }
 
+// Migrate the legacy state TaskStateFinish for group task to TaskStateFailed|TaskStatePartialFailed|TaskStateAllSuccess
+func (s *Service) migrateLegacyState(ctx context.Context) {
+	var groupTasks []TaskGroupModel
+	var tasks []TaskModel
+	for {
+		// Step 1: find out group tasks whose state is TaskStateFinish
+		err := s.params.LocalStore.Where("state = ?", TaskStateFinish).Order("id DESC").Limit(100).Find(&groupTasks).Error
+		if err != nil || len(groupTasks) == 0 {
+			break
+		}
+
+		for _, groupTask := range groupTasks {
+			// Step 2: find out all child tasks of the group task
+			err = s.params.LocalStore.Where("task_group_id = ?", groupTask.ID).Find(&tasks).Error
+			if err != nil || len(tasks) == 0 {
```

Comment:
This condition should never happen, but if it happens, may cause the outer loop to become a dead loop, still consider how to avoid it.

Reply:
Add way: Add `.Where("id < ?", lastMinID")` condition when querying group tasks.